PluginCode/src/main/java/smr/cs/ualberta/libcomp/DatabaseAccess.java

```diff
@@ -481,16 +484,67 @@ public void updateUserProfile(User userRecord) throws IOException {
 
         if (Integer.parseInt(userRecord.getCloudStore()) == 1) {
             sendUser(userRecord.getUserID(), obj.toString());
+
+            sendFeedbackToCloud(); // check if the user has local feedback stored previously, send them now.
         }
     }
 
-    public void updateFeedback(ReplacementFeedback replacementFeedback) throws IOException {
+    public void storeLocalFeedback(JSONObject obj) throws IOException {
+
+        String filePath = this.filePath +"\\Feedback.json";
+        File myFile = new File(filePath);
+        JSONObject Mainobj = null; // = new JSONObject();
+        if (myFile.exists()) {
+            String content = FileUtils.readFileToString(myFile, "utf-8");
+            JSONObject objM = new JSONObject(content);
+            JSONArray jsonarr = objM.getJSONArray("Feedbacks");
+            jsonarr.put(obj);
+            Mainobj = new JSONObject();
+            Mainobj.put("Feedbacks", jsonarr);
+        } else
+        {
+            Mainobj = new JSONObject();
+            JSONArray array = new JSONArray();
+            array.put(obj);
+            Mainobj.put("Feedbacks", array);
+        }
+
+        FileOutputStream fOuts = new FileOutputStream(myFile);
+        String result = Mainobj.toString();
+        OutputStreamWriter myOutWriter = new OutputStreamWriter(fOuts);
+        myOutWriter.append(result);
+        myOutWriter.close();
+    }
+
+    public void sendFeedbackToCloud() throws IOException {
+        String filePath = this.filePath +"\\Feedback.json";
+        String tokenValue = getUserToken();
+        File myFile = new File(filePath);
+        boolean successfulsent = false;
+        if (myFile.exists()) {
```

Comment:
Ok, so same comment for everything about names.. and be consistent in variable name style (camel case)

Reply:
renamed to feedbackFile